client/platform/desktop/backend/platforms/common.ts

```diff
@@ -63,6 +61,13 @@ async function getDatasetBase(datasetId: string): Promise<{
     return false;
   });
 
+  const jsonFileCandidates = await contents
+    .filter((name) => JsonFileName.test(name))
+    .filter(async (name) => {
+      const statResult = await fs.stat(npath.join(datasetFolderPath, name));
+      return statResult.isFile();
+    });
```

Comment:
I'm having some issues with this, I'm guessing an async filter might need to be done differently?  Something like creating a map of all the promises and awaiting on them and then filtering once they're complete?  From my testing it looks like it doesn't await for the second filtering to be done before continuing on.